tests/functional/controllers/test_job_controller.py

```diff
@@ -0,0 +1,179 @@
+    new_user.save()
+    job_name = 'TestJob'
+    data = {'name': job_name,
+            'description': 'testDescription',
+            'userId': 1}
+
+    resp = client.post(ENDPOINT, headers=HEADERS, data=json.dumps(data))
+    resp_json = json.loads(resp.data.decode('utf-8'))
+
+    assert resp.status_code == HTTPStatus.CREATED
+    assert resp_json['job']['id'] is not None
+    assert resp_json['job']['name'] == job_name
+    assert Job.get(int(resp_json['job']['id'])) is not None
+
+
+# DELETE /jobs/{job_id}
+def test_delete_job(tables, client, new_job, new_task, new_user):
+    new_user.save()
+    new_job.save()
+    new_task.save()
+    new_job.add_task(new_task)
+    resp = client.get(BASE_URI + '/tasks?jobId={}'.format(new_job.id), headers=HEADERS)
+    resp_json = json.loads(resp.data.decode('utf-8'))
+    assert resp.status_code == HTTPStatus.OK
+    assert len(resp_json['tasks']) == 1
+
+    resp = client.delete(ENDPOINT + '/{}'.format(new_job.id), headers=HEADERS)
+
+    resp = client.get(BASE_URI + '/tasks?jobId={}'.format(new_job.id), headers=HEADERS)
+    assert resp.status_code == HTTPStatus.NOT_FOUND  # checks if task from deleted job is deleted by cascade
+
+
+# GET /tasks?job_id=1
+def test_get_tasks_from_job(tables, client, new_job, new_task, new_user):
+    new_user.save()
+    new_job.save()
+    new_task.save()
+    new_job.add_task(new_task)
+
+    resp = client.get(BASE_URI + '/tasks?jobId={}'.format(new_job.id), headers=HEADERS)
+    resp_json = json.loads(resp.data.decode('utf-8'))
+
+    assert resp.status_code == HTTPStatus.OK
+    assert len(resp_json['tasks']) == 1
+
+
+# PUT /jobs/{id}/tasks/{id}
+def test_add_task_to_job(tables, client, new_job, new_task, new_user):
+    new_user.save()
+    new_job.save()
+    new_task.save()
+
+    resp = client.put(ENDPOINT + '/{}/tasks/{}'.format(new_job.id, new_task.id), headers=HEADERS)
+
+    assert resp.status_code == HTTPStatus.OK
+    assert new_job == new_task.job
+    assert new_task in new_job.tasks
+
+
+# DELETE /jobs/{id}/tasks/{id}
+def test_remove_task_from_job(tables, client, new_job, new_task, new_user):
+    new_user.save()
+    new_job.save()
+    new_task.save()
+    new_job.add_task(new_task)
+
+    resp = client.delete(ENDPOINT + '/{}/tasks/{}'.format(new_job.id, new_task.id), headers=HEADERS)
+
+    assert resp.status_code == HTTPStatus.OK
+    assert new_task.job is None
+    assert len(new_job.tasks) == 0
+
+
+# GET /jobs/{id}/execute
+def test_execute_job(tables, client, new_job, new_user, new_task):
+    new_user.save()
+    new_job.save()
+    new_task.save()
+    new_job.add_task(new_task)
+
+    resp = client.get(ENDPOINT + '/{}/execute'.format(new_job.id), headers=HEADERS)
+    resp_json = json.loads(resp.data.decode('utf-8'))
+    assert resp.status_code == HTTPStatus.UNPROCESSABLE_ENTITY  # spawning unsuccessful because of test env (sync)
+    assert resp_json['not_spawned_list'] == [1]  # id of a task that didn't spawn successfully
```

Comment:
What is the purpose of this test?
Do we have any idea how to test the job execute endpoint? Maybe it should be tested at the level of unit tests instead of this functional test?

Reply:
Fixed in #57af3245bd78908f981b1ff0391403897b3d3314

Reply:
Fixed in #572591c